File: Exercise_1_FirstAndLast_position.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int x=-1;
        int y=-1;
        int a=-1;
        int b=-1;
        if (nums.length == 0){
            return new int []{-1,-1};
        }
        int s = 0;
        int e = nums.length-1;
        while (s<=e){
            int m = s+(e-s)/2;
            if (nums[m] == target){
                x=m;
                y=m;
                a=x;
                b=y;
                while (nums[x]==target && x > 0){
                    --x;
                }
                while (nums[y]==target && y < nums.length-1){
                    ++y;
                }
                if (nums[x]!=target){
                x++;}
                if (nums[y]!=target){
                y--;}
                return new int[]{x,y};
            }
            if(target<nums[m]){
                e=m-1;}
            else{s=m+1;}
            }
        
        
    
        return new int []{-1,-1};
}}

File: Exercise_1_FirstAndLast_position.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        res = []
        i, j = 0, len(nums)-1
        # normal binary search nothing fancy here
        while i <= j:
            mid = (i+j)//2
            if nums[mid] == target:
                # now we have found the target, just need to find start and end positions
                start, end = mid, mid
                while start > 0 and nums[start-1] == target:
                    start -= 1
                while end < len(nums)-1 and nums[end+1] == target:
                    end += 1
                return [start, end]
            elif nums[mid] > target:
                j = mid-1
            else:
                i = mid+1
        return [-1, -1]





